mods/persistence/controllers/subsystems/persistence.dm

```diff
@@ -21,15 +21,15 @@
 /datum/controller/subsystem/persistence/proc/SaveExists()
 	if(save_exists)
 		return save_exists
-	var/DBQuery/query = dbcon.NewQuery("SELECT COUNT(*) FROM `thing`;")
-	query.Execute()
-	if(query.NextRow())
-		save_exists = text2num(query.item[1]) > 0
-		in_loaded_world = save_exists
+	save_exists = serializer.save_exists()
+	in_loaded_world = save_exists
 
 /datum/controller/subsystem/persistence/proc/SaveWorld()
 	// Collect the z-levels we're saving and get the turfs!
 	to_world_log("Saving [LAZYLEN(SSpersistence.saved_levels)] z-levels. World size max ([world.maxx],[world.maxy])")
+	sleep(5)
```

Comment:
What's the purpose of these random sleeps throughout this proc?

Reply:
They're not random. They allow the messages to actually get sent before the server freezes while working. Idk if you've seen how all the message are literally pointless because they all show up at once after the save happens currently. I think its some sort of buffer flushing issue with byond.